packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -81,6 +122,38 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
       })
       .next(() => results);
   }
+
+  getNextDocumentChanges(
+    transaction: PersistenceTransaction
+  ): PersistencePromise<MaybeDocument[]> {
+    const documentPromises: Array<PersistencePromise<MaybeDocument>> = [];
+
+    let changedKeys = documentKeySet();
+
+    const range = IDBKeyRange.lowerBound(
+      this.lastReturnedKey,
+      /*lowerOpen=*/ true
+    );
+
+    return documentChangesStore(transaction)
+      .iterate({ range }, (_, snapshotChange) => {
+        changedKeys = changedKeys.unionWith(
+          this.serializer.fromDbResourcePaths(snapshotChange.changes)
+        );
+        this.lastReturnedKey = snapshotChange.autoGeneratedKey;
```

Comment:
For some reason I don't love 'autoGeneratedKey'.  I think it seems weird to put the generation mechanism into the field name...  I'd be more inclined to just name it 'id' or 'changeId' or something...  But at the same time, autoGeneratedKey is very clear and explicit and so if you like it, feel free to keep it.

Reply:
I originally called this an "id", then changes this to "changeKey", then to "autoGeneratedKey". I am fine with either. "id" is probably better for consumers of this data.